ArduPlane/quadplane.cpp

```diff
@@ -1005,6 +1005,20 @@ void QuadPlane::run_z_controller(void)
         last_pidz_init_ms = now;
     }
     last_pidz_active_ms = now;
+
+#if AP_SCRIPTING_ENABLED
+    if (thrust_override.start_ms != 0) {
```

Comment:
If this was above the "last_pidz_active_ms" reset check then you could rely on it to do the reset.

In general I'm not a fan of having this here. The method is called `run_z_controller` now not only does it not run the Z controller (which we were doing sometimes anyway) it actually runs a different controller. I would prefer to see this in `mode_guided.cpp` where instead of calling this `run_z_controller` function it does the throttle override.